crates/networking/p2p/peer_channels.rs

```diff
@@ -57,19 +62,23 @@ impl PeerChannels {
         )
     }
 
-    /// Requests block headers from the peer, starting from the `start` block hash towards newer blocks
+    /// Requests block headers from the peer, starting from the `start` block hash towards either older or newer blocks depending on the order
     /// Returns the block headers or None if:
     /// - There are no available peers (the node just started up or was rejected by all other nodes)
     /// - The response timed out
     /// - The response was empty or not valid
-    pub async fn request_block_headers(&self, start: H256) -> Option<Vec<BlockHeader>> {
+    pub async fn request_block_headers(
+        &self,
+        start: H256,
+        order: BlockRequestOrder,
+    ) -> Option<Vec<BlockHeader>> {
         let request_id = rand::random();
         let request = RLPxMessage::GetBlockHeaders(GetBlockHeaders {
             id: request_id,
             startblock: start.into(),
             limit: BLOCK_HEADER_LIMIT,
             skip: 0,
-            reverse: false,
+            reverse: matches!(order, BlockRequestOrder::NewToOld),
```

Comment:
nit: `order: BlockRequestOrder::NewToOld`?

Reply:
This is the RLPX Message that follows the p2p spec